### PR TITLE
remember all stepwise lastfailed pytest-dev/pytest#5655

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: src/_pytest/stepwise.py [hunk 1]
```diff
@@ -78,13 +81,15 @@ def pytest_runtest_logreport(self, report):
             if self.skip:
                 # Remove test from the failed ones (if it exists) and unset the skip option
                 # to make sure the following tests will not be skipped.
-                if report.nodeid == self.lastfailed:
```
Comment (graingert) [inline src/_pytest/stepwise.py]: so it turns out pytest_runtest_logreport can get called multiple times when running under xdist

**Comment by graingert** [comment]
@nicoddemus hmm this still doesn't work because xdist doesn't run log report for passing tests so it won't remove the test from lastfailed...

**Comment by graingert** [comment]
also you can end up with a situation with batched tests:
```
[1, 2, 3, 4, 5] gw0
[6, 7, 8, 9, 10] gw1
```

if 5 and 9 are both failing tests, and only tests 1, 2, 3, 6, 7, 8, 9 run (because gw0 is shutdown after 9 fails) then 5 will be marked as passing

**Comment by blueyed** [comment]
@nicoddemus why was this closed?

**Comment by blueyed** [comment]
Moved to https://github.com/pytest-dev/pytest-xdist/issues/458.

@nicoddemus please adhere to your own suggestions though (https://github.com/pytest-dev/pytest/pull/6533#issuecomment-577179396).

**Comment by nicoddemus** [comment]
Definitely, I should have left a note that this was moved to pytest-xdist. 👍

**Pull request closed without merging**